server/src/test/java/org/elasticsearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -244,65 +250,71 @@ public void testSuccessfulSnapshotAndRestore() {
 
         final int shards = randomIntBetween(1, 10);
         final int documents = randomIntBetween(0, 100);
-        TestClusterNode masterNode =
+
+        final StepListener<AcknowledgedResponse> createRepositoryListener = new StepListener<>();
+
+        final TestClusterNode masterNode =
             testClusterNodes.currentMaster(testClusterNodes.nodes.values().iterator().next().clusterService.state());
+        masterNode.client.admin().cluster().preparePutRepository(repoName)
+            .setType(FsRepository.TYPE).setSettings(Settings.builder().put("location", randomAlphaOfLength(10)))
+            .execute(createRepositoryListener);
+
+        final StepListener<CreateIndexResponse> createIndexResponseStepListener = new StepListener<>();
+        createRepositoryListener.whenComplete(acknowledgedResponse -> masterNode.client.admin().indices().create(
+            new CreateIndexRequest(index).waitForActiveShards(ActiveShardCount.ALL).settings(defaultIndexSettings(shards)),
+            createIndexResponseStepListener), SnapshotResiliencyTests::rethrowAssertion);
+
+        final StepListener<CreateSnapshotResponse> createSnapshotResponseListener = new StepListener<>();
+        createIndexResponseStepListener.whenComplete(createIndexResponse -> {
```

Comment:
1. If we are using bulk, why not to prepare the single bulk request and send it just once and avoid CountDownLatch usage?
2. After doing it, I would create separate bulkResponseListener and chain it with createSnapshotResponseListener.

Reply:
Thanks for spotting, no clue why I made it sent multiple requests here back when I created this :) -> moved to a single bulk request as suggested.

Reply:
The first item is addressed, but what about the second item? I suggest introducing new StepListener called bulkResponseListener and chain it createSnapshotResponseListener. This way you can get rid of `CountDownLatch`.

Reply:
Ah right ... the counter doesn't make much sense now either, killed it and just added an assertion on the count to be safe :)